packages/intersect-polygon/intersect-polygon.js

```diff
@@ -68,34 +85,85 @@ module.exports = (georaster, geom, run_on_values) => {
+          edges.forEach((edge, index) => {
+              let [[x1,y1], [x2,y2]] = edge;
+              let ymin = Math.min(y1, y2);
+              let ymax = Math.max(y1, y2);
+              if (target >= ymin && target <= ymax) {
+                  overlaps.push(JSON.stringify({ index, edge}));
+              }
+          }); 
+        //console.log("overlaps:", overlaps);
+      }
+
+      // iterate through the list of polygon vertices, convert them to
+      // lines, and compute the intersections with each image row
+      let intersections_by_row = _.range(num_rows).map(row => []);
+      if (debug_level >= 1) console.log("intersections_by_row.length:", intersections_by_row.length);
+      let number_of_edges = edges.length;
+      if (debug_level >= 1) console.log("number_of_edges:", number_of_edges);
+      for (let i = 0; i < number_of_edges; i++) {
+
+       
         // get vertices that make up an edge and convert that to a line
         let edge = edges[i];
-        let start_point = edge[0];
-        let end_point = edge[1];
+
+        let [start_point, end_point] = edge;
+        let [ x1, y1 ] = start_point;
+        let [ x2, y2 ] = end_point;
+
+        let direction = Math.sign(y2 - y1);
+        let horizontal = y1 === y2;
+        let vertical = x1 === x2;
+
+        let edge_y = y1;
+
         let edge_line = get_line_from_points(start_point, end_point);
 
-        let start_lng, end_lng;
-        if (start_point[0] < end_point[0]) {
-            start_lng = start_point[0];
-            end_lng = end_point[0];
+        let edge_ymin = Math.min(y1, y2);
+        let edge_ymax = Math.max(y1, y2);
+        
+        if (debug_level >= 2) {
+            console.log("\nedge", i, ":", edge);
+            console.log("direction:", direction);
+            console.log("horizontal:", horizontal);
+            console.log("vertical:", vertical);
+            console.log("edge_ymin:", edge_ymin);
+            console.log("edge_ymax:", edge_ymax);
+        }        
+
+        let start_lng, start_lat, end_lat, end_lng;
+        if (x1 < x2) {
+            [ start_lng, start_lat ] = start_point;
+            [ end_lng, end_lat ] = end_point;
         } else {
-            start_lng = end_point[0];
-            end_lng = start_point[0];
+            [ start_lng, start_lat ] = end_point;
+            [ end_lng, end_lat ]  = start_point;
         }
-        //console.log("\n\n\n");
-        //console.log("start_lng:", start_lng);
-        //console.log("end_lng:", end_lng);
+
+
+        if (start_lng === undefined) throw Error("start_lng is " + start_lng);
 
         // find the y values in the image coordinate space
-        let y_1 = Math.floor((lat_0 - start_point[1]) / cell_height);
-        let y_2 = Math.floor((lat_0 - end_point[1]) / cell_height);
+        let y_1 = Math.round((lat_0 - .5*cell_height - start_lat ) / cell_height);
+        let y_2 = Math.round((lat_0 - .5*cell_height - end_lat) / cell_height);
 
```

Comment:
? I'm a little confused as to why this changed, and why you are subtracting half the cell height